### PR TITLE
fix: truncate request vectors when reverting exec outcome

### PR DESCRIPTION
**Pull request by tcoratger**
When we apply `revert_to` to a mutable instance of `ExecutionOutcome`, the receipts are cut to the new length but the request vector should also be cut to this new length.
- Implementation is fixed.
- A unit test is modified to catch this behaviour properly.

### CODE REVIEW

File: crates/evm/execution-types/src/bundle.rs [hunk 1]
```diff
@@ -241,12 +241,16 @@ impl ExecutionOutcome {
     pub fn revert_to(&mut self, block_number: BlockNumber) -> bool {
         let Some(index) = self.block_number_to_index(block_number) else { return false };
 
+        println!("Reverting to block number: {}", index);
+
```
Comment (Rjected) [inline crates/evm/execution-types/src/bundle.rs:245]:
```suggestion
```

**Review by Rjected — CHANGES_REQUESTED** [review]
just need to remove printlns, otherwise this looks good to me

**Review by mattsse — APPROVED** [review]
nice catch!

**Review by Rjected — APPROVED** [review]
lgtm

**Pull request merged**